crates/fluvio-version-manager/src/command/switch.rs

```diff
@@ -0,0 +1,230 @@
+use crate::common::settings::Settings;
+use crate::common::workdir::{fvm_versions_path, fluvio_binaries_path};
+use crate::common::notify::Notify;
+
+/// Represents the contents of a version directory (`~/.fvm/versions/<version>`)
+/// where binaries and the manifest are stored.
+///
+/// The `VersionDirectory` represents the contents of a version directory
+/// in 2 parts:
+///
+/// - `contents`: Every file in the directory except for the manifest
+/// - `manifest`: The manifest file
+struct VersionDirectory {
+    contents: Vec<PathBuf>,
+    manifest: PathBuf,
+}
+
+impl VersionDirectory {
+    /// Opens a directory and returns a `VersionDirectory` build from scanning it
+    fn open(path: PathBuf) -> Result<Self> {
+        let mut dir = Self {
+            contents: vec![],
+            manifest: PathBuf::new(),
+        };
+
+        for entry in read_dir(path)? {
+            let entry = entry?;
+
+            if entry.metadata()?.is_dir() {
+                continue;
+            }
+
+            let entry_path = entry.path();
+
+            if entry_path.file_name().ok_or(anyhow::anyhow!(
+                "Failed to get filename from path: {}",
+                entry_path.display()
+            ))? == PACKAGE_SET_MANIFEST_FILENAME
+            {
+                dir.manifest = entry_path;
+            } else {
+                dir.contents.push(entry_path);
+            }
+        }
+
+        if dir.manifest == PathBuf::new() {
+            return Err(anyhow::anyhow!(
+                "Failed to find manifest file in version directory"
+            ));
+        }
+
+        Ok(dir)
+    }
+
+    /// Copies `contents` files into the `target` directory
+    fn copy_contents(&self, target: impl AsRef<Path>) -> Result<()> {
+        for entry in &self.contents {
+            let filename = entry.file_name().ok_or(anyhow::anyhow!(
+                "Failed to get filename from path: {}",
+                entry.display()
+            ))?;
+            let target_path = target.as_ref().join(filename);
+
+            copy(entry, &target_path)?;
+            tracing::info!(?target_path, "Copied binary");
+        }
+
+        Ok(())
+    }
+}
+
+#[derive(Debug, Parser)]
+pub struct SwitchOpt {
+    #[command(flatten)]
+    global_opts: GlobalOptions,
+    /// Version to set as active
+    #[arg(index = 1)]
+    version: Option<Channel>,
+}
+
+impl SwitchOpt {
+    pub async fn process(&self) -> Result<()> {
+        let Some(version) = self.version.clone() else {
+            self.notify_help(format!(
+                "You can use {} to see installed versions",
+                "fvm show".bold()
+            ));
+
+            return Err(anyhow::anyhow!("No version provided"));
+        };
+
+        let versions_path = fvm_versions_path()?;
+
+        // If the user runs this command before installing any versions, we
+        // should notify them and exit early.
+        if !versions_path.exists() {
```

Comment:
ditto

Reply:
Here I check if the `~/.fvm/versions` directory exists. This directory is where `fvm install` stores binaries pulled from the registry.